### PR TITLE
Improve Parallel gem RBS

### PR DESCRIPTION
**Pull request by ybiquitous**
- Make RBS definitions more accurate
- Pass the failed tests
- Renew the test script
- Update grosser/parallel submodule

The following test is now successful:

```shell
bin/test gems/parallel/1.20
```

### CODE REVIEW

File: gems/parallel/1.20/_scripts/test
Comment (ybiquitous) [inline gems/parallel/1.20/_scripts/test:1]: [note] Copied from another gem.

File: gems/parallel/1.20/_src
Comment (ybiquitous) [inline gems/parallel/1.20/_src:1]: [note] Updated to https://github.com/grosser/parallel/releases/tag/v1.23.0

File: gems/parallel/1.20/_test/Steepfile [hunk 1]
```diff
@@ -4,7 +4,7 @@ target :test do
   check "."
   signature "."
 
-  repo_path ENV["REPO_ROOT"]
+  repo_path "../../../"
```
Comment (ybiquitous) [inline gems/parallel/1.20/_test/Steepfile:7]: [note] Changed like other gems.

File: gems/parallel/1.20/_test/all.rb [hunk 2]
```diff
@@ -6,13 +6,16 @@
...
+Parallel.all?(['a','b','c'], finish: ->(item, index, result) { puts "Finish: #{item}, #{index + 1}, #{result}" }) { |item| item.size == 1 }
 
 Parallel.all?('a'..'c') { |item| item.size == 1 }
 
 items = [1,2,3]
-Parallel.all?(-> { items.pop || Parallel::Stop }) { |item| item.size == 1 }
+Parallel.all?(-> { items.pop || Parallel::Stop }) { |item|
+  # @type var item: Integer
+  item.zero?
+}
```
Comment (ybiquitous) [inline gems/parallel/1.20/_test/all.rb:19]: [note] Need to remove `_ = ` and use `@type` to pass Steep check

File: gems/parallel/1.20/parallel.rbs [hunk 3]
```diff
@@ -8,67 +8,70 @@ module Parallel
...
-  Stop: Object
+  type stop = Object
+  Stop: stop
 
   VERSION: String
 
   Version: String
 
-  def self.all?: [T] (Enumerable[T] | (^() -> T | untyped) array, # TODO: `T | Stop` should be better than `T | untyped`.
+  type callable_source[T] = ^() -> (T | stop)
```
Comment (ybiquitous) [inline gems/parallel/1.20/parallel.rbs:18]: [note] `callable_source` reduces repetition.

**Review by pocke — APPROVED** [review]
Thanks!

**Pull request merged**